core/src/main/java/com/klaviyo/coresdk/networking/NetworkBatcher.kt

```diff
@@ -0,0 +1,67 @@
+package com.klaviyo.coresdk.networking
+
+import android.os.Handler
+import android.os.HandlerThread
+import com.klaviyo.coresdk.KlaviyoConfig
+import java.util.concurrent.ConcurrentLinkedQueue
+
+object NetworkBatcher {
+    private val handlerThread = HandlerThread("Klaviyo Network Batcher")
+    private var handler: Handler? = null
+    private var batchQueue = ConcurrentLinkedQueue<NetworkRequest>()
+    private var queueInitTime = 0L
+
+    internal fun initBatcher() {
+        if (!handlerThread.isAlive) {
+            handlerThread.start()
+            handler = Handler(handlerThread.looper)
+        }
+        handler?.post(NetworkRunnable())
+    }
+
+    internal fun forceEmptyQueue() {
+        handler?.removeCallbacksAndMessages(null)
+        handler?.post(NetworkRunnable(true))
+    }
+
+    internal fun getBatchQueueSize(): Int {
+        return batchQueue.size
```

Comment:
might want to be careful using this here too, maybe just worth noting in a comment that this is O(n)

Reply:
I added this cause I wanted to expose the batch queue size easily for testing. I don't see an issue in having this exposed as is, even if its an O(n) operation